spago.dhall

```diff
@@ -0,0 +1,19 @@
+{-
+Welcome to a Spago project!
+You can edit this file as you like.
+-}
+{ name = "halogen-select"
+, dependencies =
+  [ "argonaut"
+  , "console"
+  , "debug"
+  , "effect"
+  , "halogen"
+  , "halogen-hooks"
+  , "halogen-hooks-extra"
+  , "psci-support"
+  , "record"
+  ]
+, packages = ./packages.dhall
+, sources = [ "src/**/*.purs", "test/**/*.purs" ]
+}
```

Comment:
If you'd like to switch the project to Spago, I'd like to ensure that generating a new `bower.json` file from the Spago dependencies produces a Bower file that matches the dependencies and versions currently there. That probably means that you move the test sources to a separate `test.dhall` file, and don't include dependencies like `debug`, `console`, `argonaut`, and `psci-support` (perhaps those should go in an `examples.dhall` file?) The `halogen-hooks` and `halogen-hooks-extra` are OK.

You might want to look at the migration to Spago for Halogen as a comparison:
https://github.com/purescript-halogen/purescript-halogen/pull/653

Reply:
How do I generate a bower.json file using Spago? It looks like `spago publish` is no longer a command in Spago `v0.15.2`.

Also, there is no `test` folder.

I've split up the dependencies between `spago.dhall` and `examples.dhall`

Reply:
That’s odd — I used it in the Halogen migration and I can’t find anything about it in the release notes. Without that command I don’t know how to generate one.

Reply:
D'oh! Looks like its `spago bump-version` that generates the `bower.json` file.

Reply:
Looks like it only worked for you because `halogen` was already on the Bower registry before they stopped all future uploads of new packages. Since our packages weren't already on there, this fails:
```
$ spago bump-version major
[info] Generating a new Bower config using the package set versions..
[error] Failed to run: `bower info --json purescript-halogen-hooks-extra#v0.6.0`
[error] [{
  "code": "ENOTFOUND",
  "data": {
    "endpoint": {
      "name": "",
      "source": "purescript-halogen-hooks-extra",
      "target": "v0.6.0"
    }
  },
  "id": "ENOTFOUND",
  "level": "error",
  "message": "Package purescript-halogen-hooks-extra not found",
  "stacktrace": "Error: Package purescript-halogen-hooks-extra not found\n    at createError (/home/jordan/.nvm/versions/node/v13.12.0/lib/node_modules/bower/lib/util/createError.js:4:15)\n    at /home/jordan/.nvm/versions/node/v13.12.0/lib/node_modules/bower/lib/core/resolverFactory.js:243:23\n    at _fulfilled (/home/jordan/.nvm/versions/node/v13.12.0/lib/node_modules/bower/lib/node_modules/q/q.js:854:54)\n    at /home/jordan/.nvm/versions/node/v13.12.0/lib/node_modules/bower/lib/node_modules/q/q.js:883:30\n    at Promise.promise.promiseDispatch (/home/jordan/.nvm/versions/node/v13.12.0/lib/node_modules/bower/lib/node_modules/q/q.js:816:13)\n    at /home/jordan/.nvm/versions/node/v13.12.0/lib/node_modules/bower/lib/node_modules/q/q.js:624:44\n    at runSingle (/home/jordan/.nvm/versions/node/v13.12.0/lib/node_modules/bower/lib/node_modules/q/q.js:137:13)\n    at flush (/home/jordan/.nvm/versions/node/v13.12.0/lib/node_modules/bower/lib/node_modules/q/q.js:125:13)\n    at processTicksAndRejections (internal/process/task_queues.js:79:11)"
}]

[error] Failed to run: `bower info --json purescript-halogen-hooks#v0.4.0`
[error] [{
  "code": "ENOTFOUND",
  "data": {
    "endpoint": {
      "name": "",
      "source": "purescript-halogen-hooks",
      "target": "v0.4.0"
    }
  },
  "id": "ENOTFOUND",
  "level": "error",
  "message": "Package purescript-halogen-hooks not found",
  "stacktrace": "Error: Package purescript-halogen-hooks not found\n    at createError (/home/jordan/.nvm/versions/node/v13.12.0/lib/node_modules/bower/lib/util/createError.js:4:15)\n    at /home/jordan/.nvm/versions/node/v13.12.0/lib/node_modules/bower/lib/core/resolverFactory.js:243:23\n    at _fulfilled (/home/jordan/.nvm/versions/node/v13.12.0/lib/node_modules/bower/lib/node_modules/q/q.js:854:54)\n    at /home/jordan/.nvm/versions/node/v13.12.0/lib/node_modules/bower/lib/node_modules/q/q.js:883:30\n    at Promise.promise.promiseDispatch (/home/jordan/.nvm/versions/node/v13.12.0/lib/node_modules/bower/lib/node_modules/q/q.js:816:13)\n    at /home/jordan/.nvm/versions/node/v13.12.0/lib/node_modules/bower/lib/node_modules/q/q.js:624:44\n    at runSingle (/home/jordan/.nvm/versions/node/v13.12.0/lib/node_modules/bower/lib/node_modules/q/q.js:137:13)\n    at flush (/home/jordan/.nvm/versions/node/v13.12.0/lib/node_modules/bower/lib/node_modules/q/q.js:125:13)\n    at processTicksAndRejections (internal/process/task_queues.js:79:11)"
}]
```

Reply:
Oh, shoot. Hmm. What if we add an explicit GitHub URL for these instead of pulling from the package set? We can't publish the new library to Pursuit without a valid Bower file. In the worst case we can just ensure that the Spago.dhall dependencies match the Bower dependencies, at the same major verisons, before release.

Reply:
I mean... there's always [Starsuit](https://spacchetti.github.io/starsuit/)

>  In the worst case we can just ensure that the Spago.dhall dependencies match the Bower dependencies, at the same major verisons, before release.

Meaning the `bower.json` file matches the depdencies' versions specified in the package set?

Reply:
Sorry, that's right.

Reply:
Done.